io-backend/src/main/java/agh/io/iobackend/service/StatisticsService.java

```diff
@@ -0,0 +1,118 @@
+                int p = points.get(user);
+                points.put(user, p + gameMapHistory.getPoints());
+            } else {
+                points.put(user, gameMapHistory.getPoints());
+            }
+        }
+        return points.entrySet()
+                .stream()
+                .sorted((c1, c2) -> -1 * c1.getValue().compareTo(c2.getValue()))
+                .collect(
+                        toMap(Map.Entry::getKey, Map.Entry::getValue, (e1, e2) -> e2, LinkedHashMap::new));
+    }
+
+    public UserRankResponse getUserRanks(Long userId) {
+        LinkedHashMap<User, Integer> userRanking = getUsersRanking();
+        int currentPosition = 1;
+        for (Map.Entry<User, Integer> entry : userRanking.entrySet()) {
+            if (entry.getKey().getUserId().equals(userId)) {
+                return new UserRankResponse(currentPosition, entry.getValue());
+            }
+            currentPosition += 1;
+        }
+        return null;
+    }
+
+    public LinkedHashMap<GameMap, Integer> getMapsWithTheMostWinsForUser(Long userId) {
+        List<GameMapHistory> gameMapHistories = gameMapHistoryRepository.findAll();
+        Map<GameMap, Integer> wins = new HashMap<>();
+        for (GameMapHistory gameMapHistory : gameMapHistories) {
+            if (!gameMapHistory.getUser().getUserId().equals(userId) || !gameMapHistory.isWin()) {
+                continue;
+            }
+            GameMap gameMap = gameMapHistory.getMap();
+            if (wins.containsKey(gameMap)) {
+                int p = wins.get(gameMap);
+                wins.put(gameMap, p + 1);
+            } else {
+                wins.put(gameMap, 1);
+            }
+        }
+        return wins.entrySet()
+                .stream()
+                .sorted((c1, c2) -> -1 * c1.getValue().compareTo(c2.getValue()))
+                .collect(
+                        toMap(Map.Entry::getKey, Map.Entry::getValue, (e1, e2) -> e2, LinkedHashMap::new));
+    }
+
+    public LinkedHashMap<GameMap, Integer> getMapsWithTheMostGamesForUser(Long userId) {
+        List<GameMapHistory> gameMapHistories = gameMapHistoryRepository.findAll();
+        Map<GameMap, Integer> games = new HashMap<>();
+        for (GameMapHistory gameMapHistory : gameMapHistories) {
+            if (!gameMapHistory.getUser().getUserId().equals(userId)) {
+                continue;
+            }
+            GameMap gameMap = gameMapHistory.getMap();
+            if (games.containsKey(gameMap)) {
+                int p = games.get(gameMap);
+                games.put(gameMap, p + 1);
+            } else {
+                games.put(gameMap, 1);
+            }
+        }
+        return games.entrySet()
+                .stream()
+                .sorted((c1, c2) -> -1 * c1.getValue().compareTo(c2.getValue()))
+                .collect(
+                        toMap(Map.Entry::getKey, Map.Entry::getValue, (e1, e2) -> e2, LinkedHashMap::new));
+    }
+}
```

Comment:
Domyślam się że to była główna oś tego taska. Bardzo mi się podoba, jestem fanem podejścia funkcyjnego :D